### PR TITLE
Feature/ms 06/mongoose/sessions

### PR DESCRIPTION
**Pull request by Micaela-89**
**SESSIONS ENTITY**

**Controllers added:**
getAllSessions()
getSessionById()
createNewSession()
updateSession()
deleteSession()

**Request Methods added:**
GET - /sessions
GET - /sessions/:id
POST - /sessions
PUT - /sessions/:id
DELETE - /sessions/:id

### CODE REVIEW

**Comment by Micaela-89** [comment]
SESSIONS ENTITY

**Controllers added:**
getAllSessions()
getSessionById()
createNewSession()
updateSession()
deleteSession()

**Request Methods added:**
GET - /sessions
GET - /sessions/:id
POST - /sessions
PUT - /sessions/:id
DELETE - /sessions/:id

**Comment by Micaela-89** [comment]
SESSIONS ENTITY

**Controllers added:**
getAllSessions()
getSessionById()
createNewSession()
updateSession()
deleteSession()

**Request Methods added:**
GET - /sessions
GET - /sessions/:id
POST - /sessions
PUT - /sessions/:id
DELETE - /sessions/:id

File: models/Sessions.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const SessionSchema = new Schema(
+  {
+    id: { type: mongoose.Schema.Types.ObjectId },
+    postulant_id: {type: mongoose.Schema.Types.ObjectId, ref: 'Postulants'},
```
Comment (aquilesmariani) [inline models/Sessions.js:8]: I think we should replace this to be camelCase

File: postman_collection.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			"response": []
+		},
+		{
+			"name": "createNewSession",
+			"request": {
+				"method": "POST",
+				"header": [],
+				"body": {
+					"mode": "raw",
+					"raw": "{\r\n    \"date\": \"12/15/2021\",\r\n    \"time\": \"13\",\r\n    \"postulant_id\": \"6192f548af5995b361586c89\",\r\n    \"counselor_id\": \"6191930580595469c7fe1c6f\"\r\n}",
```
Comment (aquilesmariani) [inline postman_collection.json:69]: We should modify this so we can send counselorId and postulantId with camelCase

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Just small details.

Also, merge development newest changes, add the .env file with the PORT and DARABASE_URL variables as Esteban mentioned on slack.

File: controllers/sessions.js [hunk 3]
```diff
@@ -1,128 +1,72 @@
...
-        })
+const deleteSession = (req, res) => {
+  Sessions.findByIdAndDelete(req.params.id, (error) => {
+    if (!req.params.id) {
+      return res.status(400).json({ msg: `Session with id: ${req.params.id} was not found` });
+    }
+    if (error) {
+      return res.status(400).json(error);
     }
+    return res.status(200).json(`Session with id: ${req.params.id} was deleted`);
```
Comment (aquilesmariani) [inline controllers/sessions.js:62]: It will be nice if you can verify if a counselor was found before returning a 200 message

**Review by aquilesmariani — COMMENTED** [review]
Looks good @Micaela-89! You still need to merge the development branch into this one so you can fix conflicts. 

Once you do that, it will be ready to merge!

**Review by aquilesmariani — APPROVED** [review]
Looks great!

**Pull request merged**